utils/browser.py

```diff
@@ -229,7 +229,25 @@ def ensure_open(self, url_key=None):
             self.start(url_key=url_key)
         return self.browser
 
+    def add_cleanup(self, callback):
+        assert self.browser is not None
+        try:
+            cl = self.browser.__cleanup
+        except AttributeError:
+            cl = self.browser.__cleanup = []
```

Comment:
Dunder because the property is private to the manager class

The default should be a empty list if its created earlier, should I change I?

Personal preference is to leave as is